src/hotspot/share/jfr/recorder/checkpoint/types/jfrTypeSet.cpp

```diff
@@ -347,7 +347,7 @@ static void do_write_klass(JfrCheckpointWriter* writer, CldPtr cld, KlassPtr kla
   writer->write(cld != nullptr ? cld_id(cld, leakp) : 0);
   writer->write(mark_symbol(klass, leakp));
   writer->write(package_id(klass, leakp));
-  writer->write(klass->modifier_flags());
+  writer->write(klass->compute_modifier_flags());
```

Comment:
Isn't this much more expensive than grabbing the value from the mirror, especially if we have to iterate over inner classes?